lib/markdown_document.rb

```diff
@@ -0,0 +1,35 @@
+class MarkdownDocument
+  YAML_FRONT_MATTER_REGEXP = /\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)/m
+
+  attr_reader :raw_content
+
+  def initialize(raw_content)
+    @raw_content = raw_content
+  end
+
+  def front_matter
+    YAML.safe_load(@raw_content.match(YAML_FRONT_MATTER_REGEXP).to_s, permitted_classes: [Date])
```

Comment:
You may want to consider memoizing this value considering it's used multiple times in a request (e.g. for sorting the documents, then multiple times in the view for rendering individual attributes, etc.)